### ISSUE TITLE
Parallel vs Context documentation is ambiguous

### ISSUE THREAD

**Issue by asarkar** [micronaut-projects/micronaut-core#1493]
From the [Parallel Javadoc](https://docs.micronaut.io/latest/api/io/micronaut/context/annotation/Parallel.html):

> Note that beans and methods that are processed in parallel (unlike Context scoped beans) will not initially fail the startup of the ApplicationContext as they may be initialized after the ApplicationContext has started and cannot participate in StartupEvent processing. If a parallel bean fails to startup it will by default stop the ApplicationContext with an error unless shutdownOnError() set to to false.

The above verbiage is quite confusing. It says parallel doesn't fail the startup, and that makes sense since beans are lazy by default. Then it says _"if a parallel bean fails to startup it will by default stop the ApplicationContext"_. Please clarify which is true and update the Javadoc.

Also explain the difference with `@Context`.

**Comment by graemerocher** [comment]
`@Parallel` beans as noted by the annotation are an experimental feature, however the description is not inaccurate. As described in the javadoc they are started in parallel and do not block or initially fail startup. However if one does fail to load the application context is shutdown.

`@Context` scope beans are simply beans whose lifecycle is bound to that is the `ApplicationContext` They are created when the context starts and shutdown when the context stops. Unlike `@Parallel` The creation of these beans will block startup.

**Comment by asarkar** [comment]
@graemerocher "do not block or initially **fail** startup" is confusing. Perhaps a clearer wording could be "Parallel beans do not block startup, but their failure will fail the application startup."

**Comment by asarkar** [comment]
Also, I'm curious, what happens if a `@Context` scoped bean depends on one or more`@Parallel` beans? Does the initialization of the context bean wait for the initialization of the parallel beans to be completed?

**Comment by graemerocher** [comment]
No, in that case the context scope bean will cause the initialization of the `@Parallel` scoped bean early. As I said `@Parallel` is considered experimental at this time and its usage has to be dealt with with care.

**Comment by graemerocher** [comment]
Triage: Type Feature with documentation label. The current Parallel Javadoc still combines two separate facts: parallel initialization is not part of the initial StartupEvent path, and a later initialization failure can stop the context by default. It should explicitly describe startup ordering, when the parallel task is observed, the effect of shutdownOnError, and the contrast with eager Context initialization. No reproducer is needed; this is a documentation correction and clarification.
